src/cow-react/modules/swap/pure/EthFlow/EthFlowStepper/index.cosmos.tsx

```diff
@@ -350,6 +313,47 @@ const STEPS: Step[] = [
         state: SmartOrderStatus.CREATION_MINED,
         isExpired: true,
       },
+      creation: { ...defaultProps.creation, failed: false },
+    },
+  },
+  {
+    description: '[TX-FAILED-CREATION] Creation tx failed',
+    props: {
+      ...defaultProps,
+      order: {
+        ...defaultOrderProps,
+        state: SmartOrderStatus.CREATING,
+        isExpired: false,
+        rejectedReason: 'Transaction failed',
+      },
+      creation: { ...defaultProps.creation, failed: true },
+    },
+  },
+  {
+    description: '[TX-FAILED-CANCELLATION] Cancellation tx failed - refund succeeded',
+    props: {
+      ...defaultProps,
+      order: {
+        ...defaultOrderProps,
+        state: SmartOrderStatus.INDEXED,
+        isExpired: false,
+      },
+      creation: { ...defaultProps.creation, failed: false },
+      cancellation: { hash: TX, failed: true },
+      refund: { failed: false },
+    },
+  },
+  {
+    description: '[TX-FAILED-REFUND] Refund tx failed (can it?)',
+    props: {
+      ...defaultProps,
+      order: {
+        ...defaultOrderProps,
+        state: SmartOrderStatus.INDEXED,
+        isExpired: false,
+      },
+      creation: { ...defaultProps.creation, failed: false },
+      refund: { hash: TX, failed: true },
```

Comment:
This state was added, and could happen, but I don't think is well represented. Why if the refund failed we show that it is refunded.

<img width="1314" alt="image" src="https://user-images.githubusercontent.com/2352112/209009594-18c859be-4b77-4c83-8722-7033d4ba2d37.png">

Reply:
Yeah, I didn't really handled this because I don't know what should happen in that case.
Is it possible the refund tx from the backend to fail?